### ISSUE TITLE
No module named ecto

### ISSUE THREAD

**Issue by arturrz** [OpenDroneMap/odm_vagrant#21]
I already installed odm and when trying to run the following happens: 
Traceback (most recent call last):
  File "/home/artu/Documents/OpenDroneMap/run.py", line 8, in <module>
    import ecto
ImportError: No module named ecto

I checked my paths and that should not be a problem. Any Idea??

**Comment by dakotabenjamin** [comment]
Use `./run.sh` instead of `python run.py` All other parameters the same.

**Comment by Jltxar** [comment]
I have been chasing this problem since October.  Using Bash on Windows. Ubuntu 16.04.  ./run.sh just directs the program back to run.py and the "no module named ecto" error is repeated.  I have installed ODM repeatedly, using git and otherwise, worked with PATHs and tried all "fixes" that I could find.  I really would like to use the program, but it has been really frustrating.

**Comment by smathermather** [comment]
I'm confused, @Jltxar -- you are using Bash on Windows? I don't know if anyone has successfully done that. Have you tried docker? That's the recommended approach for Windows.

**Comment by Jltxar** [comment]
Thanks for the reply.

I knew nothing about Linux until I wanted to process drone images.  Made it
work on a side by side Ubuntu installation, but it seemed to me that it
permanently tied up a lot of my hard drive.  After fiddling with Docker a
bit I decided Bash on Windows looked easier. This is first item I've seen
questioning its usage.

**Comment by smathermather** [comment]
And... it may work fine, it’s just I’d be suspect if it worked with no changes to the installer, as you’re the first I know to try it.

**Comment by smathermather** [comment]
Ah, no I’m confused a bit here. You’re doing this based on vagrant. My apologies. @dakotabenjamin — are we retiring the vagrant repo in favor of docket?

Regardless @jltxar — the recommended  approach for Windows is docker, not vagrant.

**Comment by Jltxar** [comment]
You are way over my head now.  I have been trying to use Ubuntu installed
on Bash for Windows.  Don't know vagrant, unless they are identical.  I'm
strictly having to do cook book Ubuntu, no working knowledge.

**Comment by Jltxar** [comment]
I notice that this error is one that is not uncommon, and has apparently
been hard to deal with.  That had given me reason to consider the way the
program was installed.  I guess that's what help pages are for.  I'll go
back and try harder with Docker, after I hang around a bit and see if
something else comes down from this.

Thanks.

**Comment by Jltxar** [comment]
Hmm.  Hadn't mentioned I did get it to run one time.  Also had trouble
getting QGIS to work on the results and by the time I got back to ODM, I
had forgotten the magic formula.

**Comment by dakotabenjamin** [comment]
OpenDroneMap will not work in Bash for windows. Sorry if there's been any confusion. I recommend you check out our [docs](https://docs.opendronemap.org) for official ways to install ODM. If you need help, please reach out at http://community.opendronemap.org, since this repository isn't monitored.

**Comment by Jltxar** [comment]
Dakota,

I really appreciate your having taken time to help me.

Never-the-less,  I have a folder that contains the output of one drone
flight generated by ODM installed in Bash on Windows.  Wish I knew what I
twerked that made it possible, but I have yet to duplicate it.

**Comment by pierotofy** [comment]
This is not documented, but we can run ODM directly on Windows 10 using the Linux Subsystem (I've tried it a few weeks ago and it works).

![image](https://user-images.githubusercontent.com/1951843/38094541-09f9394c-333c-11e8-91a5-afb3c4fc52bd.png)

**Comment by dakotabenjamin** [comment]
Really? That's good news. I could not get it to run on my local machine.

**Comment by Jltxar** [comment]
Dakota,

I finally did it.  I have successfully run another drone flight through an
Ubuntu OpenDroneMap installation into Bash for Windows.

Again, I am not completely sure what allowed it due to days of following
help blogs suggestions, juggling paths and entering direct commands, but
the final recoverable bashrc entered paths include the two pythonpath
entries and the LD_LIBRARY_PATH entries, as instructed in the installation
steps, with the added line directing the pythonpath to
"....../python2.7/dist-packages/ecto".  Also due to the fumbling of a 78 yr
old elephant in a china shop, all of those entries are somehow duplicated.

The final change that effected the success was the command "sudo ./run.sh
--i".  I added the prefix "./" after seeing you detail that command in one
of your suggestions in an older thread.  I had never entered the run.sh
command with the "./" or the second dash before the "i".  I am almost sure
that I didn't do any of this on the previous successful run, so there must
be other factors involved.

I did close Bash on Windows, then try rerunning without making any
additional direct commands or changes in bashrc and it did work again.

Thanks again for the help,
